api/src/main/java/org/apache/iceberg/catalog/SchemaIdentifier.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.catalog;
+
+/**
+ * Identifies a schema in iceberg catalog
+ */
+public class SchemaIdentifier {
+  private final String schemaName;
+
+  public SchemaIdentifier(String schemaName) {
```

Comment:
I think this should support a multi-level namespace. We don't know how catalogs will be structured and don't want to impose a fixed number of levels. Just wrapping a string array would work.

Reply:
Done.

Reply:
@rdblue  Are you thinking of a 3 level namespace [schema, database and table] or arbitrary levels?